### PR TITLE
Schema hash enum

### PR DESCRIPTION
**Pull request by sandreae**
Introduces `SchemaId` as a new type used by the operation `schema` field.

## 📋 Checklist

- [x] Add tests that cover your changes
- [x] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [x] Link this PR to any issues it closes
- [x] New files contain a SPDX license header
- [x] Check if descriptions and terminology match `handbook` content (and visa-versa) 
  - https://github.com/p2panda/handbook/pull/148

### CODE REVIEW

**Comment by sandreae** [comment]
Think this works well. Only thing I'm not sure about is the name.

`let application_schema = SchemaType::new(DEFAULT_SCHEMA_HASH)`

Doesn't give you a schema type, it gives you the hash for a specific schema wrapped in a type..... It would be nice just to call it `Schema` but this will be taken later. Maybe `SchemaHash`?

**Comment by adzialocha** [comment]
Was stumbling upon the same thing, `SchemaHash` sounds better :+1:

File: p2panda-rs/src/document/document.rs
Comment (adzialocha) [inline p2panda-rs/src/document/document.rs:458]: Grand Type-Tennis :yellow_heart:

Comment (sandreae) [reply p2panda-rs/src/document/document.rs:458]: lolz

**Comment by adzialocha** [comment]
Think we need a final decision on this one here before we can merge this: https://github.com/p2panda/p2panda/issues/211

**Comment by sandreae** [comment]
Ah yes... true... Shame we have to actually make decisions as well as implementing them :roll_eyes:

**Comment by adzialocha** [comment]
> Ah yes... true... Shame we have to actually make decisions as well as implementing them

Finally makes us make decisions :-D

I'm fine with this current solution :+1: Could be nice to have slightly more fun hashes maybe? Like: `SOLIDIFIED0000SOFA000...0001`

File: p2panda-rs/src/document/document.rs
Comment (adzialocha) [reply p2panda-rs/src/document/document.rs:458]: Seriously! This makes so much sense and the API becomes clearer (like "wtf what hash do I use here for what?")

Comment (sandreae) [reply p2panda-rs/src/document/document.rs:458]: haha, I know, right?? I think we were waiting for some hash typing this whole time :rocket:

**Comment by sandreae** [comment]
`5C4E3A000000000000000000000000000000000000000000001`

SCHEMA = 5C4E3A (kind of)

**Comment by cafca** [comment]
I could really use this struct for the CDDL stuff. Can I help out here? With the handbook maybe?

I think `SchemaId` would be a good name.

**Comment by sandreae** [comment]
Hey! Yes, a little help would be great. I can work on the PR to get it ready now, if you don't mind doing the handbook updates that would be cool (sorry, you shouldn't always do the handbook :sob:)

Just checking, but do we already want to change `SchemaId` to contain an array of graph tips in this PR? I'm not sure.

**Comment by cafca** [comment]
I would prefer we make the change for application schemas in another pr. When we do so many interdependent things in parallel its helpful to separate concerns so nothing is blocked :)

**Comment by sandreae** [comment]
Great, I just suggested that in the chat as well ;-p

**Comment by sandreae** [comment]
But I will change application schema variant to contain a `RelationList`, right?

**Comment by sandreae** [comment]
Ughh, I'm confused, that's not a relation list, it's a DocumentViewId, right?? I think my brain is tied up in knots....

**Comment by cafca** [comment]
Also have a knot situation over here and unfortunately have to take a break now and do some other stuff :S

**Comment by cafca** [comment]
I will be adding changes to the handbook in this pr

https://github.com/p2panda/handbook/pull/148

**Comment by sandreae** [comment]
So I changed `SchemaId::Apllication(Relation)` to contain a relation. But I basically ignored that we can include an array of graph tips to identify a specific view (it _is_ optional) as I wonder if this needs to be dealt with in several places and would be better done in a different PR.

I think this is_"OK"_, let me know what you think.

**Review by adzialocha — APPROVED** [review]
Wohoo!

**Comment by cafca** [comment]
This was not ready to merge without the handbook changes.

**Pull request merged**
